L1Trigger/L1CaloTrigger/plugins/L1NNCaloTauProducer.cc

```diff
@@ -0,0 +1,914 @@
+Class: l1tNNCaloTauProducer
+Frinedly name: The TauMinator
+
+\class l1tNNCaloTauProducer l1tNNCaloTauProducer.cc
+
+Description: 
+Perform reconstruction and identification of tau 
+candidates at L1 Trigger with a CNN.
+
+Implementation:
+Take as input the HCAL TPs, the ECAL TPs from
+l1tEGammaClusterEmuProducer, and the HGCAL TPs
+from l1tHGCalTowerProducer and l1tHGCalBackEndLayer2Producer.
+Proceed to clustering of trigger towers in NxM
+clusters, match to HGcal 3D clusters in the endcap.
+Finally apply the CNNs.
+
+Original Author: Jona Motta
+Created: Tue May 30th 2023
+
+*/
+
+#include <iostream>
+#include <vector>
+#include <cmath>
+
+#include "boost/property_tree/ptree.hpp"
+#include "boost/property_tree/json_parser.hpp"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/L1TCalorimeterPhase2/interface/CaloTower.h"
+#include "DataFormats/HcalDigi/interface/HcalDigiCollections.h"
+#include "DataFormats/L1THGCal/interface/HGCalMulticluster.h"
+#include "DataFormats/L1TParticleFlow/interface/PFCluster.h"
+#include "DataFormats/L1THGCal/interface/HGCalTower.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/L1Trigger/interface/Tau.h"
+
+#include "CalibFormats/CaloTPG/interface/CaloTPGTranscoder.h"
+#include "CalibFormats/CaloTPG/interface/CaloTPGRecord.h"
+
+#include "L1Trigger/L1THGCal/interface/backend/HGCalTriggerClusterIdentificationBase.h"
+#include "L1Trigger/Phase2L1ParticleFlow/interface/HGC3DClusterEgID.h"
+#include "L1Trigger/L1TCalorimeter/interface/CaloTools.h"
+
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+struct NNmodels_GlobalCache {
+  std::string CNNmodel_CB_path;
+  std::string DNNident_CB_path;
+  std::string DNNcalib_CB_path;
+
+  std::string CNNmodel_CE_path;
+  std::string DNNident_CE_path;
+  std::string DNNcalib_CE_path;
+  std::string FeatScaler_CE_path;
+  boost::property_tree::ptree FeatScaler_CE;
+
+  tensorflow::GraphDef* CNNmodel_CB;
+  tensorflow::GraphDef* DNNident_CB;
+  tensorflow::GraphDef* DNNcalib_CB;
+
+  tensorflow::Session* CNNmodel_CBsession;
+  tensorflow::Session* DNNident_CBsession;
+  tensorflow::Session* DNNcalib_CBsession;
+
+  tensorflow::GraphDef* CNNmodel_CE;
+  tensorflow::GraphDef* DNNident_CE;
+  tensorflow::GraphDef* DNNcalib_CE;
+
+  tensorflow::Session* CNNmodel_CEsession;
+  tensorflow::Session* DNNident_CEsession;
+  tensorflow::Session* DNNcalib_CEsession;
+};
+
+class l1tNNCaloTauProducer : public edm::stream::EDProducer<edm::GlobalCache<NNmodels_GlobalCache>> {
```

Comment:
Class names should start with capital letter. Also the file name should match the module class name.